File: front-end/friendfinder/src/model/comment-response-vm.ts
import { AccountVm } from './account-vm';
import {PostResponse} from "./post-response";

export class CommentResponseVm {
  id?: number;
  content?: string;
  post?: PostResponse;
  createdBy?: string;
  updatedDate?: Date;
  createdDate?: Date;
  account?: AccountVm;

  // Pre-calculated fields for performance
  timeAgo?: string;
  isMine?: boolean;

  constructor(
    id?: number,
    content?: string,
    post?: PostResponse,
    createdBy?: string,
    updatedDate?: Date,
    createdDate?: Date,
    account?: AccountVm,
  ) {
    this.id = id;
    this.content = content;
    this.post = post;
    this.createdBy = createdBy;
    this.updatedDate = updatedDate;
    this.createdDate = createdDate;
    this.account = account;
  }
}

File: back-end/src/main/java/com/spring/boot/social/vm/CommentResponseVm.java
package com.spring.boot.social.vm;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.spring.boot.social.dto.PostDto;
import com.spring.boot.social.vm.auth.AccountVm;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CommentResponseVm {
    private Long id;
    private String content;
    private PostDto post;
    private AccountVm account;
    private String createdBy;
    private LocalDateTime updatedDate;
    private LocalDateTime createdDate;
}